Content/Items/Vitric/Weapons.CoachGunUpgrade.cs

```diff
@@ -0,0 +1,436 @@
+                    Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.Glow>(),
+                                        velo, 0, new Color(255, 150, 50), 0.85f);
+                }
+
+                Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.CoachGunDustFour>(), (velocity.RotatedByRandom(MathHelper.ToRadians(10f))) * Main.rand.NextFloat(0.3f, 0.50f)).scale = 0.8f;
+            }
+
+            for (int i = 0; i < 8; i++)
+            {
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(25, 25), ModContent.DustType<Dusts.Glow>(),
+                    Main.rand.NextVector2Circular(4, 4), 0, new Color(255, 150, 50), 0.95f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(25, 25), ModContent.DustType<Dusts.CoachGunDust>(),
+                    Main.rand.NextVector2Circular(8, 8), 70 + Main.rand.Next(60), default, Main.rand.NextFloat(1.5f, 1.9f)).rotation = Main.rand.NextFloat(6.28f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(25, 25), ModContent.DustType<Dusts.CoachGunDustTwo>(),
+                    Main.rand.NextVector2Circular(8, 8), 80 + Main.rand.Next(40), default, Main.rand.NextFloat(1.5f, 1.9f)).rotation = Main.rand.NextFloat(6.28f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(25, 25), ModContent.DustType<Dusts.CoachGunDustFour>()).scale = 0.9f;
+            }
+
+            for (int i = 0; i < 15; i++)
+            {
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(5, 5), ModContent.DustType<Dusts.Glow>(),
+                    (velocity.RotatedByRandom(MathHelper.ToRadians(15f))) * Main.rand.NextFloat(0.2f, 0.5f), 0, new Color(255, 150, 50), 0.95f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(5, 5), ModContent.DustType<Dusts.CoachGunDust>(),
+                    (velocity.RotatedByRandom(MathHelper.ToRadians(40f))) * Main.rand.NextFloat(0.7f, 1.8f), 70 + Main.rand.Next(60), default, Main.rand.NextFloat(1.5f, 1.9f)).rotation = Main.rand.NextFloat(6.28f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(5, 5), ModContent.DustType<Dusts.CoachGunDustTwo>(),
+                    (velocity.RotatedByRandom(MathHelper.ToRadians(40f))) * Main.rand.NextFloat(0.7f, 1.8f), 80 + Main.rand.Next(40), default, Main.rand.NextFloat(1.5f, 1.9f)).rotation = Main.rand.NextFloat(6.28f);
+
+                Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(5, 5), ModContent.DustType<Dusts.CoachGunDustFour>(),
+                    (velocity.RotatedByRandom(MathHelper.ToRadians(35f))) * Main.rand.NextFloat(0.15f, 0.35f)).scale = 0.9f;
+            }
+        }
+    }   
+
+    public class VitricBombDust : ModDust
```

Comment:
This file is getting long, it might be smart to seperate this dust and the buff out to their own files